hydrocron/api/controllers/timeseries.py

```diff
@@ -156,103 +350,45 @@ def add_units(gdf, columns):
-    Bad Request if any errors are found alongside 0 hits.
+def get_response(results, hits, elapsed, return_type, output):
+    """Create and return HTTP response based on results.
 
-    :param feature: Data requested for Reach or Node or Lake
-    :type feature: str
-    :param feature_id: ID of the feature to retrieve
-    :type feature_id: str
-    :param start_time: Start time of the timeseries
-    :type start_time: str
-    :param end_time: End time of the timeseries
-    :type end_time: str
-    :param output: Format of the data returned
+    :param results: Dictionary of SWOT timeseries results
+    :type results: dict
+    :param hits: Number of results returned from query
+    :type hits: int
+    :param elapsed: Number of seconds it took to query for results
+    :type elapsed: float
+    :param return_type: Accept request header
+    :type return_type: str
+    :param output: Output to return in request
     :type output: str
-    :param fields: List of requested columns
-    :type fields: dict
-
-    :rtype: dict, integer
-    """
-
-    data = {'http_code': '200 OK'}
-    if feature not in ('Node', 'Reach'):
-        data['http_code'] = '400 Bad Request'
-        data['error_message'] = f'400: feature parameter should be Reach or Node, not: {feature}'
-
-    elif not feature_id.isdigit():
-        data['http_code'] = '400 Bad Request'
-        data['error_message'] = f'400: feature_id cannot contain letters: {feature_id}'
-
-    elif not is_date_valid(start_time) or not is_date_valid(end_time):
-        data['http_code'] = '400 Bad Request'
-        data['error_message'] = '400: start_time and end_time parameters must conform to format: YYYY-MM-DDTHH:MM:SSZ or YYYY-MM-DDTHH:MM:SS-00:00'
-
-    elif output not in ('csv', 'geojson'):
-        data['http_code'] = '400 Bad Request'
-        data['error_message'] = f'400: output parameter should be csv or geojson, not: {output}'
-
-    elif not is_fields_valid(feature, fields):
-        data['http_code'] = '400 Bad Request'
-        data['error_message'] = '400: fields parameter should contain valid SWOT fields'
-
-    return data, 0
-
-
-def is_date_valid(query_date):
-    """
-    Check if the query date conforms to the correct format.
-
-    :param start_time: Start or end time of the timeseries
-    :type start_time: str
-
-    :rtype: bool
-    """
-
-    try:
-        datetime.datetime.strptime(query_date, "%Y-%m-%dT%H:%M:%S%z")
-        return True
-    except ValueError:
-        return False
-
 
-def is_fields_valid(feature, fields):
+    rtype: dict
     """
-    Check if fields are present in either the reach or node list of columns
 
-    :param fields: List of requested columns
-    :type fields: dict
+    if results['http_code'] == '200 OK':
 
-    :rtype: bool
-    """
+        if return_type in ('text/csv', 'application/geo+json'):
+            data = results['response']
+
+        else:  # 'application/json'
```

Comment:
I think the one use case that's missing from this logic is if accept = application/json and output = csv. In that case we want the json metadata but the results.csv should have content and results.geojson would be blank

Reply:
This use case is covered by the logic by the else block as the CSV results are returned for 'application/json' requests if output is set to "csv".

In other words, line 385: `data['results'][output] = results['response']` sets the response based on the value of the `output` variable.

Let me know if there is a better or clearer way to handle this.

Reply:
ah ok, yeah re-reading the code I see that now; thanks for verifying